tuner/tuner/candidate_gen.py

```diff
@@ -527,6 +634,51 @@ def tune(
         tune_logger.info(f"Configurations .pkl is stored in {output}/configs.pkl")
 
 
+def generate_configs_and_td_specs(
+    input_module: ir.Module,  # Path to the mlir file to be tuned
+    tuner_context: TunerContext,
+    limit: int = 4096,  # Max candidates to be generated
+    num_subgroups: int = 4,  # GPU spec, used to determine candidate generation constraints
+) -> list[ir.Module]:
+    dispatch_tuner_registry = DispatchTunerRegistry()
+    dispatch_tuner_registry.register(
+        [
+            ContractionOpInterfaceTuner(),
+            ConvolutionOpInterfaceTuner(),
+        ]
+    )
+
+    walk_result: OpWalkResult = walk_mlir_op(input_module, dispatch_tuner_registry)
+
+    dispatch_tuner = walk_result.dispatch_tuner
+    assert dispatch_tuner, "No suitable dispatch tuner found"
+    problem_size: ProblemSize = dispatch_tuner.get_shapes(str(input_module))
+    tune_logger.debug(str(problem_size))
+
+    # Index 0 is reserved for default config, so it gets no td spec.
+    with ir.Location.unknown() as loc:
+        empty_module = ir.Module.create(loc)
+    config_specs: list[ir.Module] = [empty_module]
```

Comment:
To avoid this special case, we could also look up the lowering_config and translation_info from the original module and re-materialize it here, but I don't think it's worth it...

Reply:
Or we could generate a placeholder spec that never matches anything.

Reply:
I kind of like the empty module for now. It seems relatively clean to me, and it makes it clear to someone looking through the specs that it is a placeholder. Having an explicit placeholder spec would also be good.